src/test/java/net/sf/jabref/importer/fileformat/MedlineImporterTest.java

```diff
@@ -0,0 +1,132 @@
+        for (String str : list) {
+            try (InputStream is = MedlineImporterTest.class.getResourceAsStream(str);) {
+                Assert.assertTrue(medlineImporter.isRecognizedFormat(is));
+            }
+        }
+    }
+
+    @Test
+    public void testIsRecognizedFormatReject() throws Throwable {
+        List<String> list = Arrays.asList("CopacImporterTest1.txt", "CopacImporterTest2.txt", "IEEEImport1.txt",
+                "IsiImporterTestWOS.isi", "oai2.xml", "oai22.xml", "RisImporterTest1.ris");
+        for (String str : list) {
+            try (InputStream is = MedlineImporterTest.class.getResourceAsStream(str);) {
+                Assert.assertFalse(medlineImporter.isRecognizedFormat(is));
+            }
+        }
+    }
+
+    @Test
+    public void testIsRecognizedFormatEmptyFile() throws IOException {
+        Assert.assertFalse(medlineImporter.isRecognizedFormat(emptyFileStream));
+    }
+
+    @Test
+    public void testImportMalformedEntry() throws IOException {
+        try (InputStream is = MedlineImporter.class.getResourceAsStream("MedlineImporterMalformedEntryTest.xml")) {
+            medlineImporter.importEntries(is, new OutputPrinterToNull());
+        }
+    }
+
+    @Test
+    public void testImportEntryWithArticleID() throws IOException {
+        checkImportEntriesAgainstBibFileEntries("MedlineImporterArticleIDTest.xml",
+                "MedlineImporterArticleIDTestBib.bib");
+        try (InputStream shouldBeIs = MedlineImporter.class.getResourceAsStream("MedlineImporterArticleIDTest.xml");
+                InputStream bibIn = MedlineImporter.class.getResourceAsStream("MedlineImporterArticleIDTestBib.bib")) {
+            List<BibEntry> entries = medlineImporter.importEntries(shouldBeIs, new OutputPrinterToNull());
+            Assert.assertEquals(1, entries.size());
+            BibEntry entry = entries.get(0);
+            BibtexEntryAssert.assertEquals(bibIn, entry);
+        }
+    }
+
+    @Test
+    public void testImportSingleEntry() throws IOException {
+        checkImportEntriesAgainstBibFileEntries("MedlineImporterSingleEntryTest.xml",
+                "MedlineImporterSingleEntryTestBib.bib");
+    }
+
+    @Test
+    public void testEntriesWithMinimalFields() throws IOException {
+        checkImportEntriesAgainstBibFileEntries("MedlineImporterMinimalEntryTest.xml",
+                "MedlineImporterMinimalEntryTestBib.bib");
+    }
+
+    private void checkImportEntriesAgainstBibFileEntries(String medlineFile, String bibTexFile) throws IOException {
+        try (InputStream is = MedlineImporter.class.getResourceAsStream(medlineFile);
+                InputStream nis = MedlineImporter.class.getResourceAsStream(bibTexFile)) {
+            List<BibEntry> entries = medlineImporter.importEntries(is, new OutputPrinterToNull());
+            Assert.assertNotNull(entries);
+            Assert.assertEquals(1, entries.size());
+            BibtexEntryAssert.assertEquals(nis, entries.get(0));
```

Comment:
See above. 
Assert.assertNotNull is not necessary as well if you use `assertEquals(InputStram, List<BibEntry>)`